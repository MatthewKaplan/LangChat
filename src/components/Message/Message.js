import React from 'react';
import { View, Text } from 'react-native';

const Message = props => {
	const { username, text } = props;
	return (
		<View>
			<Text>
				{username}: {text}
			</Text>
		</View>
	);
};

export default Message;
