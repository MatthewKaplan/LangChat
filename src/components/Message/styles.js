import { StyleSheet } from 'react-native';

const lightgrey = '#a7a7a7';
const darkgrey = '#323232';

export default StyleSheet.create({
	container: {
		paddingTop: 13,
		paddingBottom: 13,
		flexDirection: 'row',
		justifyContent: 'flex-start',
		alignContent: 'center',
		borderTopWidth: 1,
		borderColor: '#f3f3f3'
	},
	username: {
		fontWeight: '500',
		fontSize: 16,
		color: darkgrey
	},
	initialContainer: {
		width: 35,
		height: 35,
		marginRight: 15,
		marginTop: 1,
		borderRadius: 100,
		borderWidth: 1,
		borderColor: lightgrey,
		justifyContent: 'center'
	},
	initials: {
		fontSize: 20,
		textAlign: 'center',
		color: lightgrey
	},
	timestamp: {
		color: lightgrey,
		fontWeight: '300',
		marginLeft: 10
	},
	messageHeader: {
		justifyContent: 'flex-start',
		alignItems: 'flex-end',
		flexDirection: 'row'
	}
});
