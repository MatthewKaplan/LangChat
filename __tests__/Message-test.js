// __tests__/Message-test.js
import React from 'react';
import { Message } from '../src/components/Message/Message';
import { shallow } from 'enzyme';

const mockUserName = [ 'mockUser' ];
const mockMessage = "This is a mock message";
const mockSetReferencedMessage = jest.fn();

describe('Message', () => {
	let wrapper, instance;

	beforeEach(() => {
		wrapper = shallow(
			<Message username={mockUserName} message={mockMessage} setReferencedMessage={mockSetReferencedMessage}/>
		);
		instance = wrapper.instance();
	});

	it(`matches snapshot`, () => {
		expect(wrapper).toMatchSnapshot();
	});
	
	it("should invoke 'setReferencedMessage' with correct params when user goes to edit a message", () => {
		wrapper.find("[data-test='edit-mode']").simulate('LongPress')
		expect(mockSetReferencedMessage).toHaveBeenCalled();
	})
});
