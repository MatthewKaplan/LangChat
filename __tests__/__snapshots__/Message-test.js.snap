// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 100,
        "borderWidth": 1,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 15,
        "width": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      MOCKUSER
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        mockUser
      </Text>
    </View>
    <View
      style={
        Object {
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "fontSize": 16,
          }
        }
      >
        This is a mock message
      </Text>
    </View>
  </View>
</View>
`;
